webboot/webboot.go

```diff
@@ -234,12 +243,33 @@ func main() {
 
 	}
 	if *dryrun == false {
-		bookmark[filename].Cmdline = string(cl) + " " + bookmark[filename].Cmdline
+		k, i := wbpath(tmp, b.Kernel), wbpath(tmp, b.Initrd)
+		cmdline := string(cl) + " " + b.Cmdline
 
-		if err := mountkexec.KexecISO(bookmark[filename], tmp); err != nil {
-			log.Fatalf("Error in kexecISO:%v", err)
+		kernel, err := os.Open(k)
+
+		if err != nil {
+			log.Fatal(err)
+		}
+		var image boot.OSImage
+		if err := multiboot.Probe(kernel); err == nil {
+			image = &boot.MultibootImage{
+				Cmdline: cmdline,
+			}
+		} else {
+			image = &boot.LinuxImage{
+				Kernel:  uio.NewLazyFile(k),
+				Initrd:  uio.NewLazyFile(i),
+				Cmdline: cmdline,
+			}
+		}
+		if err := image.Load(true); err != nil {
+			log.Fatalf("error failed to kexec into new kernel:%v", err)
+		}
+		if err := kexec.Reboot(); err != nil {
+			log.Fatalf("error failed to Reboot into new kernel:%v", err)
 		}
 	}
 
-	fmt.Printf("The URL requested: %v\n The file requested: %v\n The mounting point: %v\n", URL, filename, tmp)
+	fmt.Printf("URL: %v\n Distro: %v\nmount point: %v\n", b.URL, b, tmp)
```

Comment:
why have each part be on separate lines? what’s wrong with having it all on one line?

Reply:
it's showing one removed, one added; it's not two separate lines.

Reply:
oh I see what you mean now, oops. 
if you run it you can see. It's just more readable.

Reply:
ah, I see what you mean, now. 
We did it that way to make it more readable. If you try it you'll see.